infrastructure/async/src/testFixtures/java/tech/pegasys/teku/infrastructure/async/Waiter.java

```diff
@@ -34,8 +35,19 @@ public class Waiter {
 
   public static void waitFor(
       final Condition assertion, final int timeoutValue, final TimeUnit timeUnit) {
-    Awaitility.waitAtMost(timeoutValue, timeUnit)
-        .ignoreExceptions()
+    waitFor(assertion, timeoutValue, timeUnit, true);
+  }
+
+  public static void waitFor(
+      final Condition assertion,
+      final int timeoutValue,
+      final TimeUnit timeUnit,
+      final boolean ignoreExceptions) {
+    ConditionFactory conditionFactory = Awaitility.waitAtMost(timeoutValue, timeUnit);
+    if (ignoreExceptions) {
+      conditionFactory = conditionFactory.ignoreExceptions();
+    }
```

Comment:
Is this actually the same as the `catchUncaughtException` option you were using previously? Or more specifically, did you confirm the test still fails if your fix isn't applied?  If so, looks good, carry on and merge.  Otherwise totally fine to just add a completely separate method here in Waiter.

Reply:
Yup confirmed the test fails without the fix. `catchUncaughtExceptions` is true by default for awaitility unless `ignoreExceptions` is used